extensions/amp-auto-lightbox/0.1/amp-auto-lightbox.js

```diff
@@ -434,23 +419,20 @@ export function runCandidates(ampdoc, candidates) {
  * Scans a document on initialization to lightbox elements that meet criteria.
  * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
  * @param {!Element=} opt_root
- * @return {!Promise<!Array<!Promise>|undefined>}
+ * @return {!Array<!Promise>|undefined}
  */
 export function scan(ampdoc, opt_root) {
-  const candidates = Scanner.getCandidates(opt_root || ampdoc.win.document);
-
-  return resolveIsEnabledForDoc(ampdoc, candidates).then(isEnabled => {
-    if (!isEnabled) {
-      dev().info(TAG, 'disabled');
-      return;
-    }
-    return runCandidates(ampdoc, candidates);
-  });
+  if (!isEnabledForDoc(ampdoc)) {
+    dev().info(TAG, 'disabled');
+    return;
+  }
+  const root = opt_root || ampdoc.win.document;
+  return runCandidates(ampdoc, Scanner.getCandidates(root));
 }
 
 
 AMP.extension(TAG, '0.1', ({ampdoc}) => {
-  ampdoc.whenReady().then(() => {
+  ampdoc.whenBodyAvailable().then(() => {
```

Comment:
This fires as soon as `body` is opened. Does not mean rest of the DOM is parsed yet. You wanna keep it as `whenReady`

Reply:
Ah, I had it the other way around! Thanks, PTAL.